lib/ansible/modules/cloud/azure/azure_rm_lock.py

```diff
@@ -0,0 +1,209 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2019 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_lock
+version_added: "2.9"
+short_description: Manage Azure locks.
+description:
+    - Create, delete an Azure lock.
+options:
+    name:
+        description:
+            - Name of the lock.
+        required: true
+    resource_id:
```

Comment:
better naming? resource_id like the id of resource in this module, such as lock resource id. resource_id_applied?